### PR TITLE
Adds rings

### PR DESCRIPTION
**Pull request by Alberyk**
Adds rings, ported from baystation12. You can make them out of most materials and etc. Also, it adds support for wearing rings under gloves, like magboots.

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -215,6 +215,8 @@ BLIND     // can't see anything
 	var/obj/item/weapon/cell/cell = 0
 	var/clipped = 0
 	var/fingerprint_chance = 0
+	var/obj/item/clothing/ring/ring = null		//Covered ring
+	var/mob/living/carbon/human/wearer = null	//Used for covered rings when dropping
```
Comment (Lohikar) [inline code/modules/clothing/clothing.dm]: This is cleared on drop, or on destroy, correct?

Comment (Alberyk) [reply code/modules/clothing/clothing.dm]: It is cleared on drop.

File: code/datums/uplink/stealthy and inconspicuous weapons.dm [hunk 2]
```diff
@@ -33,3 +33,8 @@
 	name = "Concealed Cane Sword"
 	item_cost = 9
 	path = /obj/item/weapon/cane/concealed
+
+/datum/uplink_item/item/stealthy_weapons/sleey_ring
```
Comment (Lohikar) [inline code/datums/uplink/stealthy and inconspicuous weapons.dm]: `sleey_ring`

File: code/game/atoms.dm [hunk 3]
```diff
@@ -229,9 +229,10 @@
 		//Deal with gloves the pass finger/palm prints.
 		if(!ignoregloves)
 			if(H.gloves && H.gloves != src)
-				var/obj/item/clothing/gloves/G = H.gloves
-				if(!prob(G.fingerprint_chance))
-					return 0
+				if(istype(H.gloves, /obj/item/clothing/gloves))
```
Comment (Lohikar) [inline code/game/atoms.dm]: Why not just replace 
```dm
if(H.gloves && H.gloves != src)
    if(istype(H.gloves, /obj/item/clothing/gloves))
``` 
with 
```dm
if(istype(H.gloves, /obj/item/clothing/gloves) && H.gloves != src)
```
?

File: code/modules/clothing/rings/material.dm [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	. = ..(mapload, "platinum")
+
+/obj/item/clothing/ring/material/phoron/Initialize(var/mapload)
+	. = ..(mapload, "phoron")
+
+/obj/item/clothing/ring/material/bronze/Initialize(var/mapload)
+	. = ..(mapload, "bronze")
+
+/obj/item/clothing/ring/material/glass/Initialize(var/mapload)
+	. = ..(mapload, "glass")
```
Comment (Lohikar) [inline code/modules/clothing/rings/material.dm]: what, no uranium ring?

File: code/modules/clothing/rings/material.dm [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	. = ..(mapload, "gold")
+
+/obj/item/clothing/ring/material/platinum/Initialize(var/mapload)
+	. = ..(mapload, "platinum")
+
+/obj/item/clothing/ring/material/phoron/Initialize(var/mapload)
+	. = ..(mapload, "phoron")
+
+/obj/item/clothing/ring/material/bronze/Initialize(var/mapload)
+	. = ..(mapload, "bronze")
```
Comment (Lohikar) [inline code/modules/clothing/rings/material.dm]: This is apparently a placeholder material, does this work?

File: code/modules/clothing/rings/rings.dm [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+//Sleepy Ring
+/obj/item/clothing/ring/reagent/sleepy
+	name = "silver ring"
+	desc = "A ring made from what appears to be silver."
+	icon_state = "material"
+	origin_tech = list(TECH_MATERIAL = 2, TECH_ILLEGAL = 5)
+
+/obj/item/clothing/ring/reagent/sleepy/Initialize()
+	. = ..()
+	reagents.add_reagent(/datum/reagent/chloralhydrate, 15)
```
Comment (Lohikar) [inline code/modules/clothing/rings/rings.dm]: Reagent ID must be a string, not a type.

File: code/modules/clothing/rings/material.dm (hunk 5 above)
Comment (Alberyk) [reply code/modules/clothing/rings/material.dm]: It works just fine.

File: code/modules/clothing/rings/material.dm (hunk 4 above)
Comment (Alberyk) [reply code/modules/clothing/rings/material.dm]: You can make them with materials.

File: code/datums/uplink/stealthy and inconspicuous weapons.dm (hunk 2 above)
Comment (Alberyk) [reply code/datums/uplink/stealthy and inconspicuous weapons.dm]: Fixed.

File: code/modules/clothing/rings/rings.dm (hunk 6 above)
Comment (Alberyk) [reply code/modules/clothing/rings/rings.dm]: Fixed.

File: html/changelogs/alberyk-rings.yml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+author: Alberyk
+delete-after: True
+changes: 
+  - rscadd: "Adeed rings, ported from baystation12, to the custom loadout."
```
Comment (Lohikar) [inline html/changelogs/alberyk-rings.yml]: `Adeed`

Comment (Alberyk) [reply html/changelogs/alberyk-rings.yml]: Fixed.

**Pull request merged**